massa-models/src/slot.rs

```diff
@@ -199,10 +199,10 @@ impl Slot {
     }
 
     /// returns the maximal slot
-    pub const fn max() -> Slot {
+    pub const fn max(thread_count: u8) -> Slot {
         Slot {
             period: u64::MAX,
-            thread: u8::MAX,
+            thread: thread_count,
```

Comment:
wrong, it's `thread_count.saturating_sub(1)`

Reply:
I'll put together a pr. There are other places that don't use saturating sub... I'll bring that in there as well

Reply:
#3695

Reply:
> ...Should we use `NonZeroU8` instead for thread counts? 0 threads implies no network exists, so `NonZeroU8` should guarantee at least one thread. wherever we have `thread_count.saturating_sub(1)`, we can instead do `thread_count.get() - 1`

that can be a nice followup in the future